### DISCUSSION TITLE
How to use version determined by setuptools_scm in pyproject.toml

### DISCUSSION THREAD

**Discussion by penguinpee** [pypa/setuptools#3423] [Q&A] [answered]
I cannot find any documentation on how to configure my `pyproject.toml` file to dynamically set the version provided by setuptools_scm.

I followed their [guide](https://github.com/pypa/setuptools_scm/#pyprojecttoml-usage) to include it as a build dependency and using `python -m setuptools_scm` shows it is working as expected. Running `python -m build` also produces the desired `_version.py` in the project directory. But how to get that version into `pyproject.toml`, so the package will be versioned accordingly? That bit seems to be missing from the documentation.

**Comment by abravalheri** [comment, marked as answer]
Hi @penguinpee , are you using the `[project]` table for project metadata in `pyproject.toml `?

If that is the case, you can set `dynamic = ["version"]` and leave the `version` field unspecified.

(I am assuming that you are already using the `[tool.setuptools_scm]` table for the plugin settings)

**Reply by penguinpee** [reply]
Yes, I'm using the `[project]` table as you suggested [earlier](https://github.com/pypa/setuptools/discussions/3379#discussioncomment-2964223) :wink:. With `dynamic  = ["version"]` it's working. Thank you very much again. :handshake:

Were are the tables documented? Or `pyproject.toml` for that matter. I might have used the wrong search terms, but nothing useful turned up. Anyway, the `dynamic  = ["version"]` part should definitely be included in the `setuptools_scm` README.

**Reply by abravalheri** [reply]
The setuptools specific tables can be found in https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html.

The `[project]` is part of the a larger PyPA interoperability spec, documented in https://packaging.python.org/en/latest/specifications/declaring-project-metadata/.
